index.js

```diff
@@ -2,7 +2,130 @@ const menuBox = document.getElementById('menuBox');
+     <a href="#" class="desktop-pic" ><img class="project-pics" src="${item.image}" alt="tonic Project image"></a>
+    <div class="flow">
+     <h3 class="project-title">${item.name}</h3>
+     <ul class="info">
+         <li class="conopy-li">
+             <h3 class="canopy">${item.client[0]}</h3>
+         </li>
+         <li class="couter-image">
+                 <img src="./images/Counter.png" alt=" ">
+         </li>
+         <li class="back-end-li">
+             <h3 class="back-end">${item.client[1]}</h3>
+         </li>
+         <li class="couter-image">
+                 <img src="./images/Counter.png" alt=" ">
+         </li>
+         <li>
+             <h3 class="year">${item.client[2]}</h3>
+         </li>
+     </ul> 
+ <p class="info-about">${item.dicsription}</p>
+     <ul class="info-2">
+         <li>
+             <button type="submit" class="button-1">${item.technologies[0]}</button>
+                 
+         </li>
+         <li>
+             <button type="submit" class="button-2">${item.technologies[1]}</button>
+                 
+             
+         </li>
+         <li>
+             <button type="submit" class="button-3">${item.technologies[2]}</button>
+                 
+             
+         </li>
+     </ul>
+         <button type="submit" class="button-4">See Project</button>
+   
+ </div>
+ 
+ </div>
+ </article>`;
+});
+document.querySelectorAll('.button-4').forEach((n, index) => {
+  n.addEventListener('click', () => {
+    title.innerText = arr[index].name;
+    dis.innerText = arr[index].dicsription;
+    imge.src = arr[index].image;
+    livelink.onclick = arr[index].live_version;
+    githublink.onclick = arr[index].source;
+    overlay.style.display = 'block';
+  });
+});
 
+clientsNav.innerHTML += `<div id="clients">
+<ul class="info">
+    <li class="conopy-li">
+        <h3 class="canopy">${arr[0].client[0]}</h3>
+    </li>
+    <li class="couter-image">
+            <img class="couter-image" src="./images/Counter.png" alt=" ">
+    </li>
+    <li class="back-end-li">
+        <h3 class="back-end">${arr[1].client[1]}</h3>
+    </li>
+    <li class="couter-image">
+            <img class="couter-image" src="./images/Counter.png" alt=" ">
+    </li>
+    <li>
+        <h3 class="year">${arr[2].client[2]}</h3>
+    </li>
+</ul>
+</div>`;
+unorder.innerHTML += `<div id="overlay-ul">
+<ul class="info-3">
+    <li>
+        <button type="submit" class="button-1">${arr[0].technologies[0]}</button>
+            
+    </li>
+    <li>
+        <button type="submit" class="button-2">${arr[1].technologies[1]}</button>
+            
+        
+    </li>
+    <li>
+        <button type="submit" class="button-3">${arr[3].technologies[2]}</button>
+            
+        
+    </li>
+</ul>
+</div>`;
+
+document.querySelector('#cancel').addEventListener('click', () => {
+  overlay.style.display = 'none';
+});
```

Comment:
- [ ] Please refactor your code by adding an id to every object in your array and use for loop to display data by using an index because this will make your code DRY, which stands for 'don't repeat yourself,' is a principle of software development that aims at reducing the repetition of patterns and code duplication in favor of abstractions and avoiding redundancy. :smile: 

You can follow the example below  Note that it is just an example :arrow_down: 

```
const projects = [
  {
    id: 1,
    title: 'Multi-Post Stories Gain+Glory',
    img: 'assets/images/placeholder-image.png',
    description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea',
    demoLink: 'https://aliatayee.github.io/portfolio-microverse/',
    tech: ['Ruby on rails', 'CSS', 'JavaScript', 'HTML'],
  },
  {
    id: 2,
    title: 'Multi-Post Stories Gain+Glory',
    img: 'assets/images/placeholder-image.png',
    description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea',
    demoLink: 'https://aliatayee.github.io/portfolio-microverse/',
    tech: ['Ruby on rails', 'CSS', 'JavaScript', 'HTML'],
  },
]


window.onload = () => {
  projects.forEach((project, index) => {
    document.getElementById('project_container').innerHTML += `<div class="img-patholder">
                    <div class="img-box"><img src="${project.img}" alt="project 1"></div>
                    <h3 class="title-project">${project.title}</h3>
                    <ul class="buttons">
                        <li>${project.tech}</li>
                    </ul>
                    <div class="see-project">
                        <button id="modal-btn" onclick="openModal(${index})">See project</button>
                    </div>
                </div>
      `;
  });
};

```